go-controller/cmd/ovnkube-trace/ovnkube-trace.go

```diff
@@ -550,60 +585,85 @@ func getOvnNamespace(coreclient *corev1client.CoreV1Client, override string) (st
 
 // Get the OVN Database URIs from the first container found in any pod in the ovn-kubernetes namespace with name "ovnkube-node"
 // Returns nbAddress, sbAddress, protocol == "ssl", nil
-func getDatabaseURIs(coreclient *corev1client.CoreV1Client, restconfig *rest.Config, ovnNamespace string) (string, string, bool, error) {
+func getDatabaseURIs(coreclient *corev1client.CoreV1Client, restconfig *rest.Config, ovnNamespace string, podInfo *PodInfo) (*PodInfo, error) {
+	podName := podInfo.OvnKubePodName
 	containerName := "ovnkube-node"
-	var err error
-
 	found := false
-	var podName string
-
-	listOptions := metav1.ListOptions{}
-	pods, err := coreclient.Pods(ovnNamespace).List(context.TODO(), listOptions)
+	pod, err := coreclient.Pods(ovnNamespace).Get(context.TODO(), podName, metav1.GetOptions{})
 	if err != nil {
-		return "", "", false, err
+		return nil, err
 	}
-	for _, pod := range pods.Items {
-		for _, container := range pod.Spec.Containers {
-			if container.Name == containerName {
-				found = true
-				podName = pod.Name
-				break
-			}
+	for _, container := range pod.Spec.Containers {
+		if container.Name == containerName {
+			found = true
+			break
 		}
 	}
 	if !found {
 		klog.V(5).Infof("Cannot find ovnkube pods with container %s", containerName)
-		return "", "", false, fmt.Errorf("cannot find ovnkube pods with container: %s", containerName)
+		return nil, fmt.Errorf("cannot find ovnkube pods with container: %s", containerName)
 	}
 	klog.V(5).Infof("Found pod '%s' with container '%s'", podName, containerName)
 
 	psCmd := "ps -eo args | grep '/usr/bin/[o]vnkube'"
 	hostOutput, hostError, err := execInPod(coreclient, restconfig, ovnNamespace, podName, containerName, psCmd, "")
 	if err != nil {
 		klog.V(5).Infof("execInPod('%s') failed with err: '%s', stderr: '%s', stdout: '%s', Pod Name '%s' \n", psCmd, err, hostError, hostOutput, podName)
-		return "", "", false, err
+		return nil, err
+	}
+	podInfo.IsInterConnect = len(regexp.MustCompile("--enable-interconnect").FindString(hostOutput)) > 0
+	if podInfo.IsInterConnect {
+		re := regexp.MustCompile(`--zone(=| )[^\s]+`)
```

Comment:
add a comment for this regex logic as well please

Reply:
done